framework/include/constraints/AutomaticMortarGeneration.h

```diff
@@ -182,7 +183,8 @@ class AutomaticMortarGeneration : public ConsoleStreamInterface
    * Compute the two nodal tangents, which are built on-the-fly.
    * @return The nodal tangents associated with the provided \p secondary_elem
    */
-  std::array<std::vector<Point>, 2> getNodalTangents(const Elem & secondary_elem) const;
+  std::array<MooseUtils::SemidynamicVector<Point, 10>, 2>
```

Comment:
I'm curious how we arrived at 10? 9 (not very different) seems like a good number because QUAD9 should be the "biggest" face we ever have

Reply:
Sure, as of now 9 should be sufficient.

Reply:
Made it 9.